application/models/user.go

```diff
@@ -556,3 +556,70 @@ func (u *User) GetPreference(key string) (*UserPreference, error) {
 
 	return &uPref, nil
 }
+
+func (u *User) CreatePreference(key, value string) (UserPreference, error) {
+	uPref := UserPreference{}
+
+	if err := u.FindByUUID(u.Uuid.String(), "id"); err != nil {
+		err := fmt.Errorf("can't create UserPreference - error finding user with uuid %s ... %v",
+			u.Uuid.String(), err)
+		return UserPreference{}, err
+	}
```

Comment:
I believe our convention is that functions with model receivers assume that the struct members contain data from a valid record. Sometimes we check that `id > 0` but other than that, I think the calling function should do the `FindByUUID`.

Reply:
OK